File: Silver_JavaScript/11047.js
// 동전 0

const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("", (firstInput) => {
  const [n, k] = firstInput.split(" ").map(Number);
  let coinList = [];
  let count = 0;
  let remainingAmount = k;

  const getCoins = (index) => {
    if (index < n) {
      rl.question("", (coinInput) => {
        coinList.push(parseInt(coinInput));
        getCoins(index + 1);
      });
    } else {
      coinList.sort((a, b) => b - a);

      for (let coin of coinList) {
        count += Math.floor(remainingAmount / coin);
        remainingAmount %= coin;
      }

      console.log(count);
      rl.close();
    }
  };

  getCoins(0);
});

File: Silver_python/11047.py
# 동전 0

n, k = map(int, input().split())
coin_list = []

for _ in range(n):
  coin_list.append(int(input()))

coin_list.sort(reverse=True)

count = 0

for coin in coin_list:
  count += k // coin
  k %= coin

print(count)
